entity-service/src/test/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepositoryTest.kt

```diff
@@ -0,0 +1,272 @@
+
+        createRelationship(
+            EntitySubjectNode(groupEntity.id),
+            EGM_CAN_READ,
+            apiaryEntity.id
+        )
+
+        createRelationship(
+            EntitySubjectNode(userEntity.id),
+            EGM_CAN_WRITE,
+            apiaryEntity.id
+        )
+
+        val availableRightsForEntities =
+            neo4jAuthorizationRepository.getAvailableRightsForEntities(
+                "urn:ngsi-ld:User:01",
+                listOf("urn:ngsi-ld:Apiary:01")
+            )
+
+        assert(availableRightsForEntities.size == 1)
+        assert(availableRightsForEntities[0].targetEntityId == "urn:ngsi-ld:Apiary:01")
+        assert(availableRightsForEntities[0].grpRight?.type == listOf(EGM_CAN_READ))
+        assert(availableRightsForEntities[0].right?.type == listOf(EGM_CAN_WRITE))
+
+        neo4jRepository.deleteEntity("urn:ngsi-ld:User:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Group:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Apiary:01")
+    }
+
+    @Test
+    fun `it should get all user's rights on several entities`() {
+        val userEntity = createEntity(
+            "urn:ngsi-ld:User:01",
+            listOf("User"),
+            mutableListOf()
+        )
+
+        val apiaryEntity = createEntity(
+            "urn:ngsi-ld:Apiary:01",
+            listOf("Apiary"),
+            mutableListOf()
+        )
+
+        val apiaryEntity2 = createEntity(
+            "urn:ngsi-ld:Apiary:02",
+            listOf("Apiary"),
+            mutableListOf()
+        )
+
+        createRelationship(
+            EntitySubjectNode(userEntity.id),
+            EGM_CAN_WRITE,
+            apiaryEntity.id
+        )
+
+        createRelationship(
+            EntitySubjectNode(userEntity.id),
+            EGM_CAN_READ,
+            apiaryEntity2.id
+        )
+
+        val availableRightsForEntities =
+            neo4jAuthorizationRepository.getAvailableRightsForEntities(
+                "urn:ngsi-ld:User:01",
+                listOf("urn:ngsi-ld:Apiary:01", "urn:ngsi-ld:Apiary:02")
+            )
+
+        assert(availableRightsForEntities.size == 2)
+        assert(
+            availableRightsForEntities.find { it.targetEntityId == apiaryEntity.id }?.right?.type == listOf(
+                EGM_CAN_WRITE
+            )
+        )
+        assert(
+            availableRightsForEntities.find { it.targetEntityId == apiaryEntity2.id }?.right?.type == listOf(
+                EGM_CAN_READ
+            )
+        )
+
+        neo4jRepository.deleteEntity("urn:ngsi-ld:User:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Apiary:01")
+        neo4jRepository.deleteEntity("urn:ngsi-ld:Apiary:02")
+    }
+
+    @Test
+    fun `it should get all user's roles`() {
+        createEntity(
+            "urn:ngsi-ld:User:01",
+            listOf("User"),
+            mutableListOf(
+                Property(
+                    name = "https://ontology.eglobalmark.com/authorization#roles",
```

Comment:
I've seen this string a couple of times, better put it into some constant value somewhere.